### PR TITLE
[leapp] improve plugin enablement and add more collected files

### PR DESCRIPTION
**Pull request by evgeni**
The `leapp-repository` package was renamed to `leapp-upgrade-elXtoY`
in [1], and while we could list all possible combinations here,
the main `leapp` package should be enough to trigger this plugin

While at it, also enable plugin when /var/lib/leapp exists, as
after an upgrade is performed, users often uninstall the leapp packages,
but it can still be beneficial for SOS to collect data about the fact
that this is a system that was upgraded (and data about the upgrade) and
/var/lib/leapp shows exactly that.
    
Also include /etc/migration-results and /var/log/leapp/answerfile in
collected files.
    
[1] https://github.com/oamg/leapp-repository/commit/924c398a003f8ada079f2819fa3305adda7dfcda
    
---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [x] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?
- [x] Are all passwords or private data gathered by this PR [obfuscated](https://github.com/sosreport/sos/wiki/How-to-Write-a-Plugin#how-to-prevent-collecting-passwords)?

### CODE REVIEW

File: sos/report/plugins/leapp.py [hunk 1]
```diff
@@ -16,10 +16,12 @@ class Leapp(Plugin, RedHatPlugin):
     short_desc = 'Leapp upgrade handling tool'
 
     plugin_name = 'leapp'
-    packages = ('leapp', 'leapp-repository')
+    packages = ('leapp',)
+    files = ('/var/lib/leapp',)
 
     def setup(self):
         self.add_copy_spec([
+            '/etc/migration-results',
```
Comment (pirat89) [inline sos/report/plugins/leapp.py:24]: does sosreport check whether the file exists? so it's not failing when the file is not present yet. just checking to be sure.

I wuold possibly extende it right away to include also:
```
/var/log/leapp/answerfile
/var/log/leapp/dnf-plugin-data.txt
```

Comment (evgeni) [reply sos/report/plugins/leapp.py:24]: it does not fail when files are absent, no :)

Comment (evgeni) [reply sos/report/plugins/leapp.py:24]: `dnf-plugin-data.txt` is already here, just scroll down a bit.
I  can add answerfile, sure

**Comment by arif-ali** [comment]
Hi @evgeni can you squash your commits please, before we can land the PR, TiA

Also, just making sure, does the answerfile, or any new files that are being collected have passwords that need obfuscating?

**Comment by evgeni** [comment]
@arif-ali I can sure, just thought it's cleaner if they are separate as they change different aspects of the plugin

And no, to the best of my knowledge, the answers file should not contain any sensitive data, right @pirat89?

File: sos/report/plugins/leapp.py [hunk 2]
```diff
@@ -16,10 +16,13 @@ class Leapp(Plugin, RedHatPlugin):
     short_desc = 'Leapp upgrade handling tool'
 
     plugin_name = 'leapp'
-    packages = ('leapp', 'leapp-repository')
+    packages = ('leapp',)
```
Comment (jcastill) [inline sos/report/plugins/leapp.py:19]: In which version of RHEL did we stop distributing leapp-repository? 
Or in other words - will this stop the plugin executing in old RHELs where migration could potentially still happen?

Comment (evgeni) [reply sos/report/plugins/leapp.py:19]: https://github.com/oamg/leapp-repository/commit/924c398a003f8ada079f2819fa3305adda7dfcda is in leapp-repository (SRPM name!) 0.15+, which is RHEL 7.9+
Pretty sure older RHEL 7.x is not supported anymore for leapp, and even if: the `leapp` package is present in any case (unless the user uninstalled it, ofc).

Comment (jcastill) [reply sos/report/plugins/leapp.py:19]: Perfect, thank you for clarifying this!

**Comment by evgeni** [comment]
squashed

**Review by arif-ali — APPROVED** [review]

**Review by TurboTurtle — APPROVED** [review]

**Pull request merged**